app/src/main/java/nl/eduvpn/app/viewmodel/ConnectionStatusViewModel.kt

```diff
@@ -123,7 +123,8 @@ class ConnectionStatusViewModel @Inject constructor(
             certValidity.value = HtmlCompat.fromHtml(context.getString(R.string.connection_certificate_status_expired), HtmlCompat.FROM_HTML_MODE_COMPACT)
             _connectionParentAction.value = ParentAction.SessionExpired
             return false
-        } else if (timeDifferenceInSeconds < 60) {
+        }
+        if (timeDifferenceInSeconds < 60) {
```

Comment:
Could you explain how this fixes the issue?
Isn't the problem that updateCertExpiry() will get called every second with negative time difference values?

Reply:
Commit cd540c1440c46d1c759dc3b299a6f69d2ea4bc2e does not change how the code works, only how it looks. The problem was that `runUpdateCertExpiryEverySecond` was called after `stopUpdateCertExpiry`. So after the `updateCertCallback` was removed in https://github.com/eduvpn/android/blob/599a22e0a3890236068cfc3a4eadda0212a3d662/app/src/main/java/nl/eduvpn/app/viewmodel/ConnectionStatusViewModel.kt#L114 or https://github.com/eduvpn/android/blob/599a22e0a3890236068cfc3a4eadda0212a3d662/app/src/main/java/nl/eduvpn/app/viewmodel/ConnectionStatusViewModel.kt#L121 it was added again in https://github.com/eduvpn/android/blob/599a22e0a3890236068cfc3a4eadda0212a3d662/app/src/main/java/nl/eduvpn/app/viewmodel/ConnectionStatusViewModel.kt#L64 The callback is now not added anymore when `updateCertExpiry` returns false.